scheduler/src/cook/compute_cluster.clj

```diff
@@ -62,6 +62,18 @@
      :mesos-framework-id framework-id
      :db-id (or cluster-entity-id (get-mesos-cluster-entity-id (d/db conn) mesos-cluster))}))
 
+(def default-cluster-name-for-legacy (atom nil))
+
+(defn get-default-cluster-name-for-legacy
+  "What cluster name to put on for legacy jobs when generating their compute-cluster."
+  []
+  @default-cluster-name-for-legacy)
+
+; A hack to store the mesos cluster name, until we refactor the code so that we support multiple clusters. In the long term future
+; this is probably replaced with a function from driver->cluster-id, or the cluster name is propagated by function arguments and
+; closed over.
+(def mesos-cluster-name-hack (atom nil))
```

Comment:
Why do we need this in addition to `get-default-cluster-name-for-legacy`?

Reply:
mesos-cluster-name-hack is a hack that will go away when we restructure the initialization. get-default-cluster-for-legacy will exist forever in order to back-patch a cluster into legacy task structures.

Reply:
Ok, I buy that. Same comment as above, though: can we replace the atom with a string read from config?

Reply:
Ok.